doc/guides/dts/howtos.rst

```diff
@@ -204,21 +204,32 @@ variable :makevar:`DTC_OVERLAY_FILE` contains a space- or semicolon-separated
 list of overlays. If :makevar:`DTC_OVERLAY_FILE` specifies multiple files, they
 are included in that order by the C preprocessor.
 
-Here are some ways to set it:
-
-1. on the cmake build command line
-   (``-DDTC_OVERLAY_FILE="file1.overlay;file2.overlay"``)
-#. with the CMake ``set()`` command in the application ``CMakeLists.txt``,
-   before including zephyr's :file:`boilerplate.cmake` file
-#. create a ``boards/<BOARD>_<revision>.overlay`` file in the application
-   folder for the current board revision. This requires that the board supports
+Devicetree overlays can be specified by the user directly, in a
+:file:`CMakeLists.txt` file, or applied automatically.
+
+Overlay files are applied according to the following scheme:
+
+1. If ``DTC_OVERLAY_FILE`` is set, the overlay file(s) in it are merged and used
+   by the application. ``DTC_OVERLAY_FILE`` can be set in various ways:
+
+   1. In :file:`CMakeLists.txt`, before calling ``find_package(Zephyr)``
```

Comment:
This is not good style; we shouldn't nest numbered lists with the same format.

```
1. hello
     1. this
     2. is a
     3. confusing list
 2. list
 ```

It's better to use something like:

```
1. ...
    1.1 ...
    1.2 ...
    1.3 ...
2. ...
```

or

```
1. ...
    i.  ...
    ii.  ...
    iii. ...
2. ...
```

etc

Reply:
just following how it's done for config files:
https://docs.zephyrproject.org/latest/guides/build/kconfig/setting.html#the-initial-configuration

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/119808744024d61be213b3a2e6e41738ebbc2111/doc/guides/build/kconfig/setting.rst#L140-L148
```
1. If ``CONF_FILE`` is set, the configuration file(s) specified in it are
   merged and used as the application configuration. ``CONF_FILE`` can be set
   in various ways:

   1. In :file:`CMakeLists.txt`, before calling ``find_package(Zephyr)``

   2. By passing ``-DCONF_FILE=<conf file(s)>``, either directly or via ``west``

   3. From the CMake variable cache
```

so judged by your comment, you propose to cleanup that too.

Reply:
> so judged by your comment, you propose to cleanup that too.

If you have time, sure, I would +1, but I'm not asking for that. I am saying I don't want to add this confusing style here to the DT docs.